Source/WebCore/html/HTMLScriptElement.cpp

```diff
@@ -86,10 +87,39 @@ void HTMLScriptElement::didFinishInsertingNode()
     ScriptElement::didFinishInsertingNode();
 }
 
+void HTMLScriptElement::setTextExported(String&& value)
+{
+    setText(WTFMove(value));
+}
+
 // https://html.spec.whatwg.org/multipage/scripting.html#dom-script-text
-void HTMLScriptElement::setText(String&& value)
+ExceptionOr<void> HTMLScriptElement::setText(std::variant<RefPtr<TrustedScript>, String>&& value)
+{
+    return setTextContent(trustedTypeCompliantString(*scriptExecutionContext(), WTFMove(value), "HTMLScriptElement text"_s));
+}
+
+ExceptionOr<void> HTMLScriptElement::setTextContent(std::optional<std::variant<RefPtr<TrustedScript>, String>>&& value)
+{
+    return setTextContent(trustedTypeCompliantString(*scriptExecutionContext(), value ? WTFMove(*value) : emptyString(), "HTMLScriptElement textContent"_s));
+}
+
+ExceptionOr<void> HTMLScriptElement::setTextContent(ExceptionOr<String> value)
 {
-    setTextContent(WTFMove(value));
+    if (value.hasException())
+        return value.releaseException();
+
+    Node::setTextContent(value.releaseReturnValue());
```

Comment:
Another way to do this is to put `using HTMLElement::setTextContent` in the class definition so we overload rather than hiding the function inherited from `Node`. Then we can just call `setTextContent` here without the `Node::` prefix.